pkg/asset/machines/openstack/worker.go

```diff
@@ -47,30 +48,25 @@ spec:
         value:
           apiVersion: openstack.cluster.k8s.io/v1alpha1
           kind: OpenStackMachineProviderConfig
-          image:
-            id: {{.Image}}
+          cloudName: {{.CloudName}}
+          cloudsSecret: "openstack-credentials"
+          image: {{.Image}}
           flavor: {{.Machine.FlavorName}}
           placement:
             region: {{.Region}}
-          subnet:
+          networks:
             filters:
-            - name: "tag:Name"
-              values:
-              - "{{.ClusterName}}-worker-*"
-          tags:
 {{- range $key,$value := .Tags}}
-            - name: "{{$key}}"
-              value: "{{$value}}"
+            - name: "tag:{{$key}}"
+              values:
+              - "{{$value}}"
```

Comment:
I'm getting:

```
I0111 12:13:56.221433       1 controller.go:114] Running reconcile Machine for ostest-worker-8rwch
E0111 12:13:56.225632       1 controller.go:160] Error checking existence of machine instance for machine object ostest-worker-8rwch; error unmarshaling JSON: while decoding JSON: json: cannot unmarshal object into Go struct field OpenstackProviderSpec.networks of type []v1alpha1.NetworkParam
```

I think this is because networks expects an array of `filter` objects.

Something like this got my actuator to create the VMs:

```yaml
networks:
- filter:
    tag: openshiftClusterID=9a6f0a69-3033-4adb-8e7a-91c8f4b75665
```

That instance is stuck in ERROR, but I think that may be unrelated (altohugh maybe this syntax isn't fully right either).

Reply:
oh, ok. So, I failed at defining the filter. Lemme fix that

Reply:
actually, did you check it had the right network attached?

Reply:
It did not. Which may have been the ERROR I was seeing. But my environment was busted for unrelated reasons, so I can't tell for sure if that was it.

I'm running it from scratch again now.